From a5e977e098aa8540bd38c8f98a9d219c38d4c578 Mon Sep 17 00:00:00 2001
From: zhuyuhua-v <yuhua.zhu@intel.com>
Date: Thu, 15 Sep 2022 06:01:14 +0000
Subject: [PATCH 18/28] Add xpu path for GRUCell (#83723)

Add xpu path for GRUCell.
We supported a new kernel named _thnn_fused_gru_cell to fuse the small ops of GRU.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/83723
Approved by: https://github.com/ezyang
---
 aten/src/ATen/native/RNN.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/aten/src/ATen/native/RNN.cpp b/aten/src/ATen/native/RNN.cpp
index c30bb6f3f4..e53e3f789b 100644
--- a/aten/src/ATen/native/RNN.cpp
+++ b/aten/src/ATen/native/RNN.cpp
@@ -800,7 +800,7 @@ struct GRUCell : Cell<Tensor, cell_params> {
       const hidden_type& hidden,
       const cell_params& params,
       bool pre_compute_input = false) const override {
-    if (input.is_cuda()) {
+    if (input.is_cuda() || input.is_xpu()) {
       TORCH_CHECK(!pre_compute_input);
       auto igates = params.matmul_ih(input);
       auto hgates = params.matmul_hh(hidden);
-- 
2.25.1

